e2e/testcases/override_reconcile_timeout_test.go

```diff
@@ -68,88 +78,55 @@ func TestOverrideReconcileTimeout(t *testing.T) {
+	expectReconcileStatus := "Timeout"
+	nomostest.WatchForObject(nt, kinds.ResourceGroup(), "root-sync", "config-management-system",
+		[]nomostest.Predicate{
+			resourceStatusEquals(idToVerify, expectActuationStatus, expectReconcileStatus),
+		})
 
-	u := &unstructured.Unstructured{}
-	u.SetGroupVersionKind(schema.GroupVersionKind{
-		Group:   "kpt.dev",
-		Kind:    "ResourceGroup",
-		Version: "v1alpha1",
-	})
-
-	err := nt.Client.Get(nt.Context, client.ObjectKey{
-		Namespace: "config-management-system",
-		Name:      "root-sync",
-	}, u)
-	if err != nil {
-		nt.T.Fatal(err)
-	} else {
-		expectActuationStatus := "Succeeded"
-		expectReconcileStatus := "Timeout"
-		err = verifyObjectStatus(u, idToVerify, expectActuationStatus, expectReconcileStatus)
-		if err != nil {
-			nt.T.Fatal(err)
-		} else {
-			nt.T.Logf("object %q has actuation and reconcile status as expected", namespaceName)
-		}
-	}
+	// Override reconcileTimeout to 5m, namespace actuation should succeed, namespace reconcile should succeed.
+	nt.MustMergePatch(rootSync, `{"spec": {"override": {"reconcileTimeout": "5m"}}}`)
+	nt.WaitForRepoSyncs()
 
 	nt.RootRepos[configsync.RootSyncName].Remove("acme/pod-1.yaml")
-	nt.RootRepos[configsync.RootSyncName].CommitAndPush(fmt.Sprintf("Remove pod/%s (never ready)", pod1Name))
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush(fmt.Sprintf("Remove pod/%s", pod1Name))
 	nt.WaitForRepoSyncs()
 
 	// Verify pod is deleted.
 	// Otherwise deletion of a managed resource will be denied by the webhook.
 	nt.MustKubectl("delete", "pods", pod1Name, "-n", namespaceName, "--ignore-not-found")
 
-	// Override reconcileTimeout to 5m, namespace actuation should succeed, namespace reconcile should succeed.
-	nt.MustMergePatch(rootSync, `{"spec": {"override": {"reconcileTimeout": "5m"}}}`)
-	nt.WaitForRepoSyncs()
-
-	// Fix ReadinessProbe to use the right port
-	pod1.Spec.Containers[0].ReadinessProbe.ProbeHandler.TCPSocket.Port =
-		intstr.FromInt(int(pod1.Spec.Containers[0].Ports[0].ContainerPort))
-
 	nt.RootRepos[configsync.RootSyncName].Add("acme/pod-1.yaml", pod1)
-	nt.RootRepos[configsync.RootSyncName].CommitAndPush(fmt.Sprintf("Add pod/%s (fixed)", pod1Name))
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush(fmt.Sprintf("Add pod/%s", pod1Name))
 	nt.WaitForRepoSyncs()
 
-	err = nt.Client.Get(nt.Context, client.ObjectKey{
-		Namespace: "config-management-system",
-		Name:      "root-sync",
-	}, u)
-	if err != nil {
-		nt.T.Error(err)
-	} else {
-		expectActuationStatus := "Succeeded"
-		expectReconcileStatus := "Succeeded"
-		err = verifyObjectStatus(u, idToVerify, expectActuationStatus, expectReconcileStatus)
-		if err != nil {
-			nt.T.Fatal(err)
-		} else {
-			nt.T.Logf("object %q has actuation and reconcile status as expected", namespaceName)
-		}
-	}
+	expectActuationStatus = "Succeeded"
+	expectReconcileStatus = "Succeeded"
+	nomostest.WatchForObject(nt, kinds.ResourceGroup(), "root-sync", "config-management-system",
```

Comment:
Do we need to Watch or Wait here?

If the sync is complete, the pod should already be scheduled and available, right?

You should be able to just use Validate (instead of WatchForObject) with the same predicates.

Reply:
Yes, Validate is good enough here, we do not need extra wait for resource status check here since we have WaitForRepoSyncs